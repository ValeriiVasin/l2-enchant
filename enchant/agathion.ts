import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { mergeEnchantMaps } from '../helpers/merge-enchants';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

const chances = [100, 100, 100, 65, 50, 35, 12, 9, 7, 4];

export const agathionEnchantMap = mergeEnchantMaps([
  createEnchantMap({
    strategy: EnchantStrategy.Destroy,
    enchantItem: Item.AgathionEnchantScroll,
    chances,
    items: itemsRange(Item.Agathion, Item.Agathion_10),
  }),
  new Map([
    [
      Item.Agathion_10,
      {
        item: Item.Agathion_10,
        cost: new Map([
          [Item.Adena, 3_000_000],
          [Item.AgathionSoulStone, 1],
        ]),
        results: [{ chance: 100, item: Item.HighQualityAgathion }],
      },
    ],
  ]),
  createEnchantMap({
    strategy: EnchantStrategy.Safe,
    enchantItem: Item.StableScrollEnchantHighQualityAgathion,
    chances: [40, 40, 40, 30, 30, 30, 20, 20, 20, 10],
    items: itemsRange(Item.HighQualityAgathion, Item.HighQualityAgathion_10),
  }),
]);

export const agathionStableEnchantMap = mergeEnchantMaps([
  createEnchantMap({
    strategy: EnchantStrategy.Destroy,
    enchantItem: Item.AgathionEnchantScroll,
    items: itemsRange(Item.Agathion, Item.Agathion_5),
    chances: chances.slice(0, 5),
  }),
  createEnchantMap({
    strategy: EnchantStrategy.Safe,
    enchantItem: Item.StableScrollEnchantAgathion,
    items: itemsRange(Item.Agathion_5, Item.Agathion_10),
    chances: chances.slice(5),
  }),
]);
