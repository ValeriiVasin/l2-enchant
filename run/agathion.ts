import { agathionStableEnchantMap } from '../enchant/agathion';
import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.Agathion, itemsRange(Item.Agathion_4, Item.Agathion_10), 100_000);
run(Item.Agathion_10, [Item.HighQualityAgathion], 1);
run(Item.HighQualityAgathion, itemsRange(Item.HighQualityAgathion_1, Item.HighQualityAgathion_10), 100_000);

console.log('-- Stable --');
run(Item.Agathion, itemsRange(Item.Agathion_1, Item.Agathion_10), 100_000, agathionStableEnchantMap);
