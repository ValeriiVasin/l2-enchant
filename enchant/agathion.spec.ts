import { agathionEnchantMap, agathionStableEnchantMap } from './agathion';

it('enchant map: agathion', () => {
  expect(agathionEnchantMap).toMatchSnapshot();
});

it('enchant map: agathion (stable)', () => {
  expect(agathionStableEnchantMap).toMatchSnapshot();
});
