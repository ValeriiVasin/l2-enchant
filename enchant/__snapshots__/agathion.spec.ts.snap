// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: agathion (stable) 1`] = `
Map {
  "Агатион" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +1",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +1" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +1",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +2",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +2" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +2",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +3",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +3" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +3",
    "results": Array [
      Object {
        "chance": 65,
        "item": "Агатион +4",
      },
      Object {
        "chance": 35,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +4" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +4",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Агатион +5",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +5" => Object {
    "cost": "Стабильный Свиток: Модифицировать Агатиона",
    "item": "Агатион +5",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Агатион +6",
      },
      Object {
        "chance": 65,
        "item": "Агатион +5",
      },
    ],
  },
  "Агатион +6" => Object {
    "cost": "Стабильный Свиток: Модифицировать Агатиона",
    "item": "Агатион +6",
    "results": Array [
      Object {
        "chance": 12,
        "item": "Агатион +7",
      },
      Object {
        "chance": 88,
        "item": "Агатион +6",
      },
    ],
  },
  "Агатион +7" => Object {
    "cost": "Стабильный Свиток: Модифицировать Агатиона",
    "item": "Агатион +7",
    "results": Array [
      Object {
        "chance": 9,
        "item": "Агатион +8",
      },
      Object {
        "chance": 91,
        "item": "Агатион +7",
      },
    ],
  },
  "Агатион +8" => Object {
    "cost": "Стабильный Свиток: Модифицировать Агатиона",
    "item": "Агатион +8",
    "results": Array [
      Object {
        "chance": 7,
        "item": "Агатион +9",
      },
      Object {
        "chance": 93,
        "item": "Агатион +8",
      },
    ],
  },
  "Агатион +9" => Object {
    "cost": "Стабильный Свиток: Модифицировать Агатиона",
    "item": "Агатион +9",
    "results": Array [
      Object {
        "chance": 4,
        "item": "Агатион +10",
      },
      Object {
        "chance": 96,
        "item": "Агатион +9",
      },
    ],
  },
}
`;

exports[`enchant map: agathion 1`] = `
Map {
  "Агатион" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +1",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +1" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +1",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +2",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +2" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +2",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион +3",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +3" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +3",
    "results": Array [
      Object {
        "chance": 65,
        "item": "Агатион +4",
      },
      Object {
        "chance": 35,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +4" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +4",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Агатион +5",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +5" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +5",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Агатион +6",
      },
      Object {
        "chance": 65,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +6" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +6",
    "results": Array [
      Object {
        "chance": 12,
        "item": "Агатион +7",
      },
      Object {
        "chance": 88,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +7" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +7",
    "results": Array [
      Object {
        "chance": 9,
        "item": "Агатион +8",
      },
      Object {
        "chance": 91,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +8" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +8",
    "results": Array [
      Object {
        "chance": 7,
        "item": "Агатион +9",
      },
      Object {
        "chance": 93,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +9" => Object {
    "cost": "Свиток: Модифицировать Агатиона",
    "item": "Агатион +9",
    "results": Array [
      Object {
        "chance": 4,
        "item": "Агатион +10",
      },
      Object {
        "chance": 96,
        "item": "Nothing",
      },
    ],
  },
  "Агатион +10" => Object {
    "cost": Map {
      "Адена" => 3000000,
      "Камень Души Агатиона" => 1,
    },
    "item": "Агатион +10",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Агатион Высокого Качества",
      },
    ],
  },
  "Агатион Высокого Качества" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Агатион Высокого Качества +1",
      },
      Object {
        "chance": 60,
        "item": "Агатион Высокого Качества",
      },
    ],
  },
  "Агатион Высокого Качества +1" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +1",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Агатион Высокого Качества +2",
      },
      Object {
        "chance": 60,
        "item": "Агатион Высокого Качества +1",
      },
    ],
  },
  "Агатион Высокого Качества +2" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Агатион Высокого Качества +3",
      },
      Object {
        "chance": 60,
        "item": "Агатион Высокого Качества +2",
      },
    ],
  },
  "Агатион Высокого Качества +3" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +3",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Агатион Высокого Качества +4",
      },
      Object {
        "chance": 70,
        "item": "Агатион Высокого Качества +3",
      },
    ],
  },
  "Агатион Высокого Качества +4" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +4",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Агатион Высокого Качества +5",
      },
      Object {
        "chance": 70,
        "item": "Агатион Высокого Качества +4",
      },
    ],
  },
  "Агатион Высокого Качества +5" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +5",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Агатион Высокого Качества +6",
      },
      Object {
        "chance": 70,
        "item": "Агатион Высокого Качества +5",
      },
    ],
  },
  "Агатион Высокого Качества +6" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +6",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Агатион Высокого Качества +7",
      },
      Object {
        "chance": 80,
        "item": "Агатион Высокого Качества +6",
      },
    ],
  },
  "Агатион Высокого Качества +7" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +7",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Агатион Высокого Качества +8",
      },
      Object {
        "chance": 80,
        "item": "Агатион Высокого Качества +7",
      },
    ],
  },
  "Агатион Высокого Качества +8" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +8",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Агатион Высокого Качества +9",
      },
      Object {
        "chance": 80,
        "item": "Агатион Высокого Качества +8",
      },
    ],
  },
  "Агатион Высокого Качества +9" => Object {
    "cost": "Стабильный Свиток Высокого Качества: Модифицировать Агатиона",
    "item": "Агатион Высокого Качества +9",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Агатион Высокого Качества +10",
      },
      Object {
        "chance": 90,
        "item": "Агатион Высокого Качества +9",
      },
    ],
  },
}
`;
